### PR TITLE
Extend Socket module

### PR DESCRIPTION
**Pull request by Lynskylate**
For #1176 Add socket method gethostname sethostname inet_aton . 
The convert_nix_error method is a common method, maybe it should be moved to exceptions.rs.

### CODE REVIEW

File: vm/src/stdlib/socket.rs [hunk 1]
```diff
@@ -372,6 +378,30 @@ fn get_addr_tuple(vm: &VirtualMachine, addr: SocketAddr) -> PyResult {
...
+
+#[cfg(unix)]
+fn socket_sethostname(hostname: PyStringRef, vm: &VirtualMachine) -> PyResult<()> {
+    sethostname(hostname.value.as_str()).map_err(|err| convert_nix_error(vm, err))
+}
+
+fn socket_inet_aton(ip_string: PyStringRef, vm: &VirtualMachine) -> PyResult {
+    match ip_string.as_str().parse::<Ipv4Addr>() {
+        Ok(ip_addr) => {
+            let out_bytes: [u8; 4] = unsafe { transmute(u32::from(ip_addr).to_be()) };
```
Comment (coolreader18) [inline vm/src/stdlib/socket.rs]: Try `u32::from(ip_addr).to_be_bytes()`, so that we can avoid `unsafe` use.

File: vm/src/stdlib/socket.rs [hunk 2]
```diff
@@ -372,6 +378,30 @@ fn get_addr_tuple(vm: &VirtualMachine, addr: SocketAddr) -> PyResult {
     Ok(vm.ctx.new_tuple(vec![ip, port]))
 }
 
+#[cfg(unix)]
+fn socket_gethostname(vm: &VirtualMachine) -> PyObjectRef {
+    let mut buf = [0u8; 1024];
+    match gethostname(&mut buf) {
+        Ok(cstr) => vm.new_str(String::from(cstr.to_str().unwrap())),
+        Err(e) => convert_nix_error(vm, e),
```
Comment (coolreader18) [inline vm/src/stdlib/socket.rs]: Did you mean to wrap these in Ok and Err? As it is now, it'll just return the exception object if there's an error.

File: vm/src/stdlib/socket.rs (hunk 1 above)
Comment (palaviv) [inline vm/src/stdlib/socket.rs]: How about using [octet](https://doc.rust-lang.org/std/net/struct.Ipv4Addr.html#method.octets)?

File: tests/snippets/stdlib_socket.py [hunk 3]
```diff
@@ -115,3 +115,9 @@
 
 with assertRaises(OSError):
 	socket.socket(socket.AF_INET, 1000)
+
+with assertRaises(OSError)	:
+	socket.inet_aton("test")
+
+assert socket.inet_aton("127.0.0.1")==b"\x7f\x00\x00\x01"
+assert socket.inet_aton("127.0.0.1")==b"\xff\xff\xff\xff"
```
[the remark this answers is not in the document]
Comment (Lynskylate) [reply tests/snippets/stdlib_socket.py]: Yeah, Thank you

> I guess you wanted "255.255.255.255"?

Thank you.

File: vm/src/stdlib/socket.rs (hunk 1 above)
Comment (Lynskylate) [reply vm/src/stdlib/socket.rs]: Great advice :). Done

File: vm/src/stdlib/socket.rs (hunk 2 above)
Comment (Lynskylate) [reply vm/src/stdlib/socket.rs]: Done

Comment (Lynskylate) [reply vm/src/stdlib/socket.rs]: Done

Comment (Lynskylate) [reply vm/src/stdlib/socket.rs]: Done

File: vm/src/stdlib/socket.rs [hunk 4]
```diff
@@ -372,6 +378,30 @@ fn get_addr_tuple(vm: &VirtualMachine, addr: SocketAddr) -> PyResult {
     Ok(vm.ctx.new_tuple(vec![ip, port]))
 }
 
+#[cfg(unix)]
+fn socket_gethostname(vm: &VirtualMachine) -> PyObjectRef {
```
[the remark this answers is not in the document]
Comment (Lynskylate) [reply vm/src/stdlib/socket.rs]: > The [gethostname](https://docs.rs/gethostname/0.2.0/gethostname/) crate is not very popular but the windows implementation looks good. What do you think about using it?

It sounds great. Does RustPython has any constraint to introduce new dependencies?

[the remark this answers is not in the document]
Comment (coolreader18) [reply vm/src/stdlib/socket.rs]: Not particularly, one of the main things we want to do with RustPython is leverage Rust's large crate ecosystem to make developing the native standard library modules; so, feel free to use it.

File: vm/src/stdlib/socket.rs [hunk 5]
```diff
@@ -372,6 +382,44 @@ fn get_addr_tuple(vm: &VirtualMachine, addr: SocketAddr) -> PyResult {
     Ok(vm.ctx.new_tuple(vec![ip, port]))
 }
 
+#[cfg(unix)]
+fn socket_gethostname(vm: &VirtualMachine) -> PyResult {
```
Comment (palaviv) [inline vm/src/stdlib/socket.rs]: You can use the same function as in windows. The `gothostname` module lib support it.

File: vm/Cargo.toml [hunk 6]
```diff
@@ -24,6 +24,7 @@ sha2 = "0.8"
 sha3 = "0.8"
 blake2 = "0.8"
 
+gethostname = "0.2.0"
```
Comment (palaviv) [inline vm/Cargo.toml]: This should not be included in WASM. You can look #1201 for example.

Comment (Lynskylate) [reply vm/Cargo.toml]: Thanks for your guidance, it helps a lot.

File: vm/src/stdlib/socket.rs (hunk 5 above)
Comment (Lynskylate) [reply vm/src/stdlib/socket.rs]: Done :)

**Comment by palaviv** [comment]
Great job @Lynskylate

**Pull request merged**